SpatialGDK/Source/SpatialGDK/Private/Interop/CrossServerRPCSender.cpp

```diff
@@ -28,7 +28,9 @@ void CrossServerRPCSender::SendCommand(const FUnrealObjectRef InTargetObjectRef,
 	CommandRequest CommandRequest(SpatialConstants::SERVER_TO_SERVER_COMMAND_ENDPOINT_COMPONENT_ID,
 								  SpatialConstants::UNREAL_RPC_ENDPOINT_COMMAND_ID);
 
-	uint64 UniqueRPCId = FMath::RandHelper(INT_MAX);
+	// We just want a semi-globally unique id for the RPC to avoid false discards.
+	FGuid Guid = FGuid::NewGuid();
+	uint64 UniqueRPCId = (((uint64)Guid.A ^ Guid.B) << 32) + (uint64)(Guid.C ^ Guid.D);
```

Comment:
Could use `CityHash64` here instead.

Reply:
I looked at it and it wants a buffer - I would feel kind of bad casting the guid to a buffer with some length.
Though the current thing is also not ideal...

Reply:
It's totally fine to cast it, the char* is just so it can do pointer arithmetic on it. Guid internally even uses it for it's hash, but casts to a uint32 first. `CityHash64((char*)&Guid, sizeof(FGuid))`

Reply:
Did it now